tiny_dnn/layers/layer.h

```diff
@@ -644,12 +717,57 @@ class layer : public node {
     initialized_ = true;
   }
 
+  /** @brief Initializes the trainable parameters of this layer.
+   *
+   * In case that a layer/node is set to be not trainable, it does
+   * nothing and returns a void. Otherwise, for each input connection
+   * and depending of the data nature (weight or bias) calls their
+   * pertinent initialization function and fill the vectors with the
+   * data generated by the mentioned functions.
+   *
+   */
+  void init_parameters() {
+    // layer/node is not trainable, do nothing and mark the layer/node
+    // as initialized.
+    if (!trainable_) {
+      initialized_ = true;
+      return;
+    }
+
+    // Fill parameter values with data generated by the initialization
+    // function. The pointer to the data is obtained from the
+    // computational graph and the methods fan_in_size() and fan_out_size()
+    // return the number of incoming/outcoming connections for each
+    // input/output unit.
+    for (size_t i = 0; i < parameters_.size(); i++) {
```

Comment:
you can use range base for loop. As for ` fan_in_size(i)` I guess we could get those from tensor shape, but not really sure what's returning